h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_7367_random_gridsearch_specific.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import division
+from __future__ import print_function
+from past.utils import old_div
+import sys
+sys.path.insert(1, "../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+from h2o.grid.grid_search import H2OGridSearch
+
+# In this test, we check to make sure that a random discrete grid search on a GAM functions correctly.
+# The test searches over 5 parameters, alpha, lambda, scale, gam_columns, and num_knots.
+# The test then compares the results of the grid search models with the models we created
+# by manually searching over the hyperspace.
+# If the coefficients do not match or an incorrect number of models is generated, the test throws an assertion error.
+class test_gam_gridsearch:
+
+    h2o_data = []
+    myX = []
+    myY = []
+    hyper_parameters = {'alpha': [0.1, 0.9], 'lambda':[0, 0.01]}
+    h2o_model = []
+    search_criteria = {'strategy': 'RandomDiscrete', "max_models": 4, "seed": 1}
+    random_hyper_params = {'alpha': [0.1, 0.5], 'lambda':[0, 0.01], 'scale': [[1, 1], [2, 2], [0.01], [0.05]], 
+                           'gam_columns': [["C6", "C7"], ["C6"]], 'num_knots': [[5, 5], [5]]};
+
+    def __init__(self):
+        self.setup_data()
+
+    def setup_data(self):
+        """
+        This function performs all initializations necessary:
+        load the data sets and set the training set indices and response column index
+        """
+        self.h2o_data = h2o.import_file(
+            path = pyunit_utils.locate("smalldata/glm_test/multinomial_10_classes_10_cols_10000_Rows_train.csv"))
+        self.h2o_data["C1"] = self.h2o_data["C1"].asfactor()
+        self.h2o_data["C2"] = self.h2o_data["C2"].asfactor()
+        self.myX = ["C1", "C2"]
+        self.myY = "C11"
+        self.h2o_data["C11"] = self.h2o_data["C11"].asfactor()
+
+    def train_models(self):
+        self.h2o_model = H2OGridSearch(H2OGeneralizedAdditiveEstimator(family="multinomial",
+                                                                       keep_gam_cols=True), hyper_params=self.random_hyper_params, search_criteria=self.search_criteria)
+        self.h2o_model.train(x = self.myX, y = self.myY, training_frame = self.h2o_data)
+        for model in self.h2o_model.models:
+            print(model.actual_params['scale'], model.actual_params['gam_columns'], model.actual_params['num_knots'])
+        print("done")
+
+def test_gridsearch():
+    test_gam_grid = test_gam_gridsearch()
+    test_gam_grid.train_models()
+
+if __name__ == "__main__":
+    h2o.init(ip='192.168.1.2', port=54321, strict_version_check=False)
```

Comment:
Karthik:  Remember to remove this before you check the files in.  It only works for you on your computer;.